### PR TITLE
feat: Add missing features to Search component

### PR DESCRIPTION
**Pull request by MoizMasud**
Signed-off-by: Moiz Masud <moizkhan_007@hotmail.com>

closes https://github.com/IBM/carbon-ui-builder/issues/19
check off Search here: https://github.com/IBM/carbon-ui-builder/issues/134

### CODE REVIEW

**Review by Akshat55 — APPROVED** [review]

File: src/fragment-components/a-searchinput.tsx [hunk 1]
```diff
@@ -15,6 +25,38 @@ export const ASearchInputSettingsUI = ({ selectedComponent, setComponent }: any)
...
+				light: checked
+			})} />
+		<Checkbox
+			labelText='AutoComplete'
+			id='autoComplete'
+			checked={selectedComponent.autoComplete === 'on'}
+			onChange={(checked: any) => setComponent({
+				...selectedComponent,
+				autoComplete: checked ? 'on' : 'off'
+			})} />
```
Comment (Akshat55) [inline src/fragment-components/a-searchinput.tsx:59]:
```suggestion
		<Checkbox
			labelText='Auto complete'
			id='autoComplete'
			checked={selectedComponent.autocomplete === 'on'}
			onChange={(checked: any) => setComponent({
				...selectedComponent,
				autocomplete: checked ? 'on' : 'off'
			})} />
```

File: src/fragment-components/a-searchinput.tsx [hunk 2]
```diff
@@ -87,19 +153,34 @@ export const ASearchInput = ({
...
-				disabled={componentObj.disabled}
-				light={componentObj.light}
-				role={componentObj.role || 'searchbox'}
-				type={componentObj.type || 'text'} />
+			{
+				componentObj.expandable
+				? <ExpandableSearch
+					size={componentObj.inputSize}
+					disabled={componentObj.disabled}
+					autoComplete={componentObj.autoComplete}
```
Comment (Akshat55) [inline src/fragment-components/a-searchinput.tsx:161]:
```suggestion
					autoComplete={componentObj.autocomplete}
```

File: src/fragment-components/a-searchinput.tsx [hunk 3]
```diff
@@ -87,19 +153,34 @@ export const ASearchInput = ({
...
+					id={componentObj.id}
+					role={componentObj.role}
+					type={componentObj.searchType} />
+				: <Search
+					size={componentObj.inputSize}
+					labelText={componentObj.label}
+					placeholder={componentObj.placeholder}
+					className={componentObj.cssClasses?.map((cc: any) => cc.id).join(' ')}
+					id={componentObj.id}
+					autoComplete={componentObj.autoComplete}
```
Comment (Akshat55) [inline src/fragment-components/a-searchinput.tsx:176]:
```suggestion
					autoComplete={componentObj.autocomplete}
```

File: src/fragment-components/a-searchinput.tsx [hunk 4]
```diff
@@ -115,32 +196,76 @@ export const componentInfo: ComponentInfo = {
 		type: 'search',
 		label: 'Search',
 		placeholder: 'Search',
-		inputSize: 'lg'
+		autoComplete: 'off',
```
Comment (Akshat55) [inline src/fragment-components/a-searchinput.tsx:199]:
```suggestion
		autocomplete: 'off',
```

File: src/fragment-components/a-searchinput.tsx [hunk 5]
```diff
@@ -115,32 +196,76 @@ export const componentInfo: ComponentInfo = {
...
+		role: 'searchbox'
 	},
 	image,
 	codeExport: {
 		angular: {
-			inputs: (_) => '',
-			outputs: (_) => '',
+			inputs: ({ json }) => `@Input() ${nameStringToVariableString(json.codeContext?.name)}Expandable = ${json.expandable};
+				@Input() ${nameStringToVariableString(json.codeContext?.name)}Placeholder = "${json.placeholder}";
+				@Input() ${nameStringToVariableString(json.codeContext?.name)}Autocomplete = "${json.autoComplete}";
```
Comment (Akshat55) [inline src/fragment-components/a-searchinput.tsx:214]:
```suggestion
				@Input() ${nameStringToVariableString(json.codeContext?.name)}Autocomplete = "${json.autocomplete}";
```

File: src/fragment-components/a-searchinput.tsx [hunk 6]
```diff
@@ -115,32 +196,76 @@ export const componentInfo: ComponentInfo = {
...
 		react: {
-			imports: ['Search'],
+			imports: ({ json }) => [json.expandable ? 'ExpandableSearch' : 'Search'],
 			code: ({ json }) => {
+				if (json.expandable) {
+					return `<ExpandableSearch
+						value={state["${json.codeContext?.name}"]}
+						size="${json.inputSize}"
+						disabled={${json.disabled}}
+						autoComplete="${json.autoComplete}"
```
Comment (Akshat55) [inline src/fragment-components/a-searchinput.tsx:244]:
```suggestion
						autoComplete="${json.autocomplete}"
```

File: src/fragment-components/a-searchinput.tsx [hunk 7]
```diff
@@ -115,32 +196,76 @@ export const componentInfo: ComponentInfo = {
...
+					onChange={handleInputChange}
+					size="${json.inputSize}"
 					labelText="${json.label}"
-					name="${json.codeContext?.name}"
 					placeholder="${json.placeholder}"
-					value={state["${json.codeContext?.name}"]}
-					${reactClassNamesFromComponentObj(json)}
-					onChange={handleInputChange} />`;
+					id="${json.id}"
+					autoComplete="${json.autoComplete}"
```
Comment (Akshat55) [inline src/fragment-components/a-searchinput.tsx:262]:
```suggestion
					autoComplete="${json.autocomplete}"
```

File: src/ui-fragment/src/components/ui-serach-input.tsx [hunk 8]
```diff
@@ -31,18 +34,33 @@ export const UISearchInput = ({ state, setState, name }: {
 		return <></>;
 	}
 
-	return <Search
-		id={state.codeContext?.name}
+	return state.expandable
+		? <ExpandableSearch
+			size={state.inputSize}
+			disabled={state.disabled}
+			autoComplete={state.autoComplete}
```
Comment (Akshat55) [inline src/ui-fragment/src/components/ui-serach-input.tsx:41]:
```suggestion
			autoComplete={state.autocomplete}
```

Also change line 10 in this file to use lowercase c so:
```
autocomplete?: string;
```

File: src/ui-fragment/src/components/ui-serach-input.tsx [hunk 9]
```diff
@@ -31,18 +34,33 @@ export const UISearchInput = ({ state, setState, name }: {
...
-		value={state.value}
-		placeholder={state.placeholder}
+		size={state.inputSize}
 		labelText={state.label}
-		autoComplete={state.autoComplete || 'off'}
-		closeButtonLabelText={state.closeButtonLabelText || 'Clear search input'}
+		placeholder={state.placeholder}
+		className={state.cssClasses?.map((cc: any) => cc.id).join(' ')}
+		id={state.codeContext?.name}
+		autoComplete={state.autoComplete}
```
Comment (Akshat55) [inline src/ui-fragment/src/components/ui-serach-input.tsx:58]:
```suggestion
		autoComplete={state.autocomplete}
```

**Review by Akshat55 — CHANGES_REQUESTED** [review]

**Review by Akshat55 — APPROVED** [review]

File: src/fragment-components/a-searchinput.tsx [hunk 10]
```diff
@@ -15,6 +25,38 @@ export const ASearchInputSettingsUI = ({ selectedComponent, setComponent }: any)
...
+			labelText='Light'
+			id='light'
+			checked={selectedComponent.light}
+			onChange={(checked: any) => setComponent({
+				...selectedComponent,
+				light: checked
+			})} />
+		<Checkbox
+			labelText='Auto complete'
+			id='autoComplete'
```
Comment (Akshat55) [inline src/fragment-components/a-searchinput.tsx:54]:
```suggestion
			id='autocomplete'
```

**Review by Akshat55 — CHANGES_REQUESTED** [review]

**Review by Akshat55 — APPROVED** [review]

**Review by zvonimirfras — APPROVED** [review]

**Pull request merged**